site/ja/guide/create_op.md

```diff
@@ -0,0 +1,1231 @@
+    .Attr("a: int >= 2");
+```
+
+* `list(<type>) >= <n>`: 長さが `<n>` 以上の、型 `<type>` のリストである。
+
+たとえば、次に示すオペレーションの登録では、アトリビュート `a` が、3つ以上の型（`int32` か `float`）のリストであることを示しています。
+
+```c++
+REGISTER_OP("TypeListExample")
+    .Attr("a: list({int32, float}) >= 3");
+```
+
+（生成されたコードでは任意である）アトリビュートのデフォルト値を設定するためには、次のように最後に `= <default>` を追加します。
+
+```c++
+REGISTER_OP("AttrDefaultExample")
+    .Attr("i: int = 0");
+```
+
+加えて、制約とデフォルト値を同時に指定することもできます。
+
+```c++
+REGISTER_OP("AttrConstraintAndDefaultExample")
+    .Attr("i: int >= 1 = 1");
+```
+
+サポートされているデフォルト値のシンタックスは、GraphDefの定義の結果として表現されるprotoで利用できるものになります。
+
+```c++
+REGISTER_OP("AttrDefaultExampleForAllTypes")
+   .Attr("s: string = 'foo'")
+   .Attr("i: int = 0")
+   .Attr("f: float = 1.0")
+   .Attr("b: bool = true")
+   .Attr("ty: type = DT_INT32")
+   .Attr("sh: shape = { dim { size: 1 } dim { size: 2 } }")
+   .Attr("te: tensor = { dtype: DT_INT32 int_val: 5 }")
+   .Attr("l_empty: list(int) = []")
+   .Attr("l_int: list(int) = [2, 3, 5, 7]");
+```
+
+特に型 `type` の値については、`tf.DType` になります。
+
+
+#### ポリモーフィズム
+
+##### 型ポリモーフィズム
+
+異なる型を入力として受け取るか、異なる型を出力するオペレーションについては、オペレーションの登録において、[入力または出力の型](#入力と出力) に [アトリビュート](#アトリビュート) を指定できます。
+一般的に、サポートされたそれぞれの型について `OpKernel` を登録します。
+
+たとえば、もし `ZeroOut` オペレーションについて、`int32` 型に加えて `float` 型をサポートするのであれば、オペレーションの登録は次のようになります。
+
+```c++
+REGISTER_OP("ZeroOut")
+    .Attr("T: {float, int32}")
+    .Input("to_zero: T")
+    .Output("zeroed: T");
+```
+
+オペレーションの登録は、どちらも型 `T` であることから、入力の型が `float` もしくは `int32` で、出力がおなじ型でなければならないと明示しています。
+
+> <a id="naming"></a>命名に関する注意点: 入力、出力、そしてアトリビュートは、スネークケースの名前にすべきです。入力の型や出力の型として使用されるアトリビュートは、例外です。これらのアトリビュートは、オペレーションがグラフに追加されるときに推測され、オペレーションの関数には現れてきません。たとえば、ZeroOutの最後の定義は、次のようなPythonの関数を生成します。
+>
+> ```python
+> def zero_out(to_zero, name=None):
+>   """...
+>   Args:
+>     to_zero: A `Tensor`. Must be one of the following types:
+>         `float32`, `int32`.
+>     name: A name for the operation (optional).
+>
+>   Returns:
+>     A `Tensor`. Has the same type as `to_zero`.
+>   """
+> ```
+>
+> もし `int32` のテンソルが `to_zero` に渡されてきた場合、`T` は自動的に `int32` （実際は、`DT_INT32`）が設定されます。これらの推測されたアトリビュートは、大文字もしくはキャメルケースで与えられます。
+>
+> ```c++
+> REGISTER_OP("StringToNumber")
+>     .Input("string_tensor: string")
+>     .Output("output: out_type")
+>     .Attr("out_type: {float, int32} = DT_FLOAT");
+>     .Doc(R"doc(
+> Converts each string in the input Tensor to the specified numeric type.
+> )doc");
+> ```
+>
+> この場合、生成されたPythonのように、ユーザは出力の型を指定しなければなりません。
+>
+> ```python
+> def string_to_number(string_tensor, out_type=None, name=None):
+>   """Converts each string in the input Tensor to the specified numeric type.
+>
+>   Args:
+>     string_tensor: A `Tensor` of type `string`.
+>     out_type: An optional `tf.DType` from: `tf.float32, tf.int32`.
+>       Defaults to `tf.float32`.
+>     name: A name for the operation (optional).
+>
+>   Returns:
+>     A `Tensor` of type `out_type`.
+>   """
+> ```
+
```

Comment:
The translation of `Type polymorphism example` should be here

Reply:
Nice catch! I fixed it.